algorithms/src/sorting/impl/Kokkos_SortImpl.hpp

```diff
@@ -206,10 +210,63 @@ void sort_onedpl(const Experimental::SYCL& space,
   auto queue  = space.sycl_queue();
   auto policy = oneapi::dpl::execution::make_device_policy(queue);
   const int n = view.extent(0);
-  oneapi::dpl::sort(policy, view.data(), view.data() + n);
+  oneapi::dpl::sort(policy, view.data(), view.data() + n,
+                    std::forward<MaybeComparator>(maybeComparator)...);
 }
 #endif
 
+template <class ExecutionSpace, class DataType, class... Properties,
+          class... MaybeComparator>
+void copy_to_host_run_stdsort_copy_back(
+    const ExecutionSpace& exec,
+    const Kokkos::View<DataType, Properties...>& view,
+    MaybeComparator&&... maybeComparator) {
+  namespace KE = ::Kokkos::Experimental;
+
+  using ViewType = Kokkos::View<DataType, Properties...>;
+  using MemSpace = typename ViewType::memory_space;
+  using layout   = typename ViewType::array_layout;
+  static_assert(
+      ViewType::rank == 1 &&
+          (std::is_same_v<typename ViewType::array_layout, LayoutRight> ||
+           std::is_same_v<typename ViewType::array_layout, LayoutLeft> ||
+           std::is_same_v<typename ViewType::array_layout, LayoutStride>),
+      "Impl::copy_to_host_run_stdsort_copy_back: supports 1D Views with "
+      "LayoutRight, LayoutLeft or LayoutStride.");
```

Comment:
why not? i think having this assert is useful to make this function kind of self-contained. so that if it is used somewhere else, we know what the constraints are

Reply:
Because it is redundant and it is a lot of code (~30% in this function).
The comment also applies to the checking accessibility below.

I am not blocking if you want to keep it though.

Reply:
removed